### PR TITLE
ensure the imagepolicyadmission plugin always has a default config

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@deads2k consider this PR a reminder for us to have a conversation about this.

**Comment by bparees** [comment]
/cc @juanvallejo 

@deads2k i assume something different needs to happen for the kubeapi server, or does this cover both?

**Comment by juanvallejo** [comment]
@bparees Thanks, this lgtm

File: pkg/image/apiserver/admission/imagepolicy/imagepolicy.go [hunk 1]
```diff
@@ -39,12 +39,14 @@ func Register(plugins *admission.Plugins) {
...
+			var config *imagepolicy.ImagePolicyConfig
 			if obj == nil {
-				return nil, nil
-			}
-			config, ok := obj.(*imagepolicy.ImagePolicyConfig)
-			if !ok {
-				return nil, fmt.Errorf("unexpected config object: %#v", obj)
+				config = &imagepolicy.ImagePolicyConfig{}
+			} else {
+				config, ok := obj.(*imagepolicy.ImagePolicyConfig)
```
Comment (juanvallejo) [inline pkg/image/apiserver/admission/imagepolicy/imagepolicy.go]:
```go
var ok bool
config, ok = obj.(*imagepolicy.ImagePolicyConfig)
```

Comment (bparees) [reply pkg/image/apiserver/admission/imagepolicy/imagepolicy.go]: le sigh.  thanks.

Comment (bparees) [reply pkg/image/apiserver/admission/imagepolicy/imagepolicy.go]: pushed.

**Comment by juanvallejo** [comment]
@bparees should we also turn this plugin on by default? https://github.com/openshift/origin/blob/master/pkg/apiserver/admission/register.go#L118

I had to remove it from that list, and add it to `DefaultOnPlugins` in order to actually get this plugin to register locally

**Comment by bparees** [comment]
@juanvallejo yes I think so, i'll update the PR.

**Comment by bparees** [comment]
updated

**Comment by juanvallejo** [comment]
Thanks, @deads2k this lgtm

**Comment by bparees** [comment]
i'm closing this as @deads2k says there's going to be a different/better way to do it post-rebase.

@juanvallejo you will also want to wait for post-rebase for your PR i think.

**Pull request closed without merging**